### PR TITLE
Log DS, joystick, and drivetrain

### PR DESCRIPTION
**Pull request by Dan-The-Alien**
This code has been tested in simulation, but not on the robot. It should log the current and temperature of each of the drive motors to a file on the RoboRIO, which we can retrieve in the pit if needed

### CODE REVIEW

**Comment by Dan-The-Alien** [comment]
Oh yeah, sorry about the settings file being included. I can pull that out into a different PR if you want, I didn't realize that file was tracked, but the only change I made is for the spotless plugin in VSCode to automatically format the file on save

File: .vscode/settings.json [hunk 1]
```diff
@@ -14,6 +14,14 @@
     "**/.factorypath": true,
     "**/*~": true
   },
+
```
Comment (CarloWoolsey) [inline .vscode/settings.json:17]: I'm fine with leaving changes to ``.vscode/settings.json`` in this commit --the changes will reduce formatting overhead for new members.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -26,6 +29,32 @@ public class Drivetrain extends SubsystemBase {
   private final DifferentialDrive m_differentialDrive =
       new DifferentialDrive(m_leftControllerGroup, m_rightControllerGroup);
 
+  private DataLog log = DataLogManager.getLog();
```
Comment (CarloWoolsey) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]:
```suggestion
  private DataLog m_log = DataLogManager.getLog();
```

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -26,6 +29,32 @@ public class Drivetrain extends SubsystemBase {
...
+  private DoubleLogEntry l_frontLeftMotorCurrent =
+      new DoubleLogEntry(log, "/custom/drivetrain/frontleft/current");
+  private DoubleLogEntry l_frontLeftMotorTemp =
+      new DoubleLogEntry(log, "/custom/drivetrain/frontleft/temp");
+  private DoubleLogEntry l_backLeftMotorCurrent =
+      new DoubleLogEntry(log, "/custom/drivetrain/backleft/current");
+  private DoubleLogEntry l_backLeftMotorTemp =
+      new DoubleLogEntry(log, "/custom/drivetrain/backleft/temp");
+  // private DoubleLogEntry l_topLeftMotorCurrent = new DoubleLogEntry(log,
```
Comment (CarloWoolsey) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: For the sake of clean code, I might suggest that we remove the commented out code blocks logging a 3rd drivetrain motor (which doesn't currently exist).

**Review by CarloWoolsey — CHANGES_REQUESTED** [review]
Looks great! I have requested a few formatting changes.

**Review by CarloWoolsey — APPROVED** [review]
Looks great!

**Pull request merged**